web/bitwarden/ciphers.go

```diff
@@ -617,3 +626,66 @@ func ShareCipher(c echo.Context) error {
 	res := newCipherResponse(cipher, setting)
 	return c.JSON(http.StatusOK, res)
 }
+
+// ImportCiphers is used to import ciphers and folders in bulk.
+func ImportCiphers(c echo.Context) error {
+	inst := middlewares.GetInstance(c)
+	if err := middlewares.AllowWholeType(c, permission.POST, consts.BitwardenCiphers); err != nil {
+		return c.JSON(http.StatusUnauthorized, echo.Map{
+			"error": "invalid token",
+		})
+	}
+
+	var req importCipherRequest
+	if err := json.NewDecoder(c.Request().Body).Decode(&req); err != nil {
+		return c.JSON(http.StatusBadRequest, echo.Map{
+			"error": "invalid JSON",
+		})
+	}
+
+	// Import the folders
+	folders := make([]interface{}, len(req.Folders))
+	olds := make([]interface{}, len(req.Folders))
```

Comment:
I don't understand the purpose of `olds` ?

Reply:
The `BulkUpdateDocs` function can be used to create documents, but also to update existing ones. And when we update an existing document, the code in stack keeps both the document before and after the change for the real-time sub-system (that is also used by triggers).

Here, we are just creating documents, so we use `nil`s value for the old documents. No purpose, except to conform to the `BulkUpdateDocs` signature.